### PR TITLE
[3006.x] Replace Slack with Discord

### LINKED ISSUE #66573: [DOCS] Link to Slack community only works if you have a saltstack.com email address
Since https://github.com/saltstack/salt/blob/master/doc/_incl/requisite_incl.rst is visible to everyone and included in many pages of public documentation it seems unhelpful to include a link to a Slack community that you can only sign up to if you have a saltstack.com email address. Is there a better community to ask and answer questions about Salt?

### PR DESCRIPTION
**Pull request by max-arnold**
### What does this PR do?

Replace Slack community links with Discord

https://www.reddit.com/r/saltstack/comments/1ecppqn/thoughts_on_the_purge_of_community_extensions/lf1smr8/

Fixes: #66573

### Merge requirements satisfied?

- [x] Docs
- [ ] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [ ] Tests written/updated

### Commits signed with GPG?
No

### CODE REVIEW

**Comment by max-arnold** [comment]
Cc: @ScriptAutomate

File: .github/ISSUE_TEMPLATE/config.yml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 blank_issues_enabled: true
 contact_links:
-  - name: Salt Community Slack
-    url: https://saltstackcommunity.slack.com/
+  - name: Salt Community Discord
+    url: https://discord.gg/YVQamSwV3g
```
Comment (ScriptAutomate) [inline .github/ISSUE_TEMPLATE/config.yml:4]:
```suggestion
    url: https://discord.gg/bPah23K7mG
```

File: .github/config.yml [hunk 2]
```diff
@@ -13,7 +13,7 @@ newIssueWelcomeComment: >
 
     - [Community Wiki](https://github.com/saltstack/community/wiki)
     - [Salt’s Contributor Guide](https://docs.saltproject.io/en/master/topics/development/contributing.html)
-    - [Join our Community Slack](https://via.vmw.com/salt-slack)
+    - [Join our Community Discord](https://discord.gg/YVQamSwV3g)
```
Comment (ScriptAutomate) [inline .github/config.yml:16]:
```suggestion
    - [Join our Community Discord](https://discord.gg/bPah23K7mG)
```

File: .github/config.yml [hunk 3]
```diff
@@ -39,7 +39,7 @@ newPRWelcomeComment: >
...
```
Comment (ScriptAutomate) [inline .github/config.yml:42]:
```suggestion
    - [Join our Community Discord](https://discord.gg/bPah23K7mG)
```

File: CONTRIBUTING.rst [hunk 4]
```diff
@@ -8,7 +8,7 @@ in a number of ways:
 -  Use Salt and open well-written bug reports.
 -  Join a `working group <https://github.com/saltstack/community>`__.
 -  Answer questions on `irc <https://web.libera.chat/#salt>`__,
-   the `community Slack <https://via.vmw.com/salt-slack>`__,
+   the `community Discord <https://discord.gg/YVQamSwV3g>`__,
```
Comment (ScriptAutomate) [inline CONTRIBUTING.rst:11]:
```suggestion
   the `community Discord <https://discord.gg/bPah23K7mG>`__,
```

File: README.rst [hunk 5]
```diff
@@ -10,9 +10,9 @@
    :alt: PyPi Package Downloads
    :target: https://lgtm.com/projects/g/saltstack/salt/context:python
 
-.. image:: https://img.shields.io/badge/slack-SaltProject-blue.svg?logo=slack
-   :alt: Salt Project Slack Community
-   :target: https://via.vmw.com/salt-slack
+.. image:: https://img.shields.io/badge/discord-SaltProject-blue.svg?logo=discord
+   :alt: Salt Project Discord Community
+   :target: https://discord.gg/YVQamSwV3g
```
Comment (ScriptAutomate) [inline README.rst:15]:
```suggestion
   :target: https://discord.gg/bPah23K7mG
```

File: README.rst [hunk 6]
```diff
@@ -180,7 +180,7 @@ used by external modules.
 A complete list of attributions and dependencies can be found here:
 `salt/DEPENDENCIES.md <https://github.com/saltstack/salt/blob/master/DEPENDENCIES.md>`_
 
-.. _Salt Project Community Slack: https://via.vmw.com/salt-slack
+.. _Salt Project Community Discord: https://discord.gg/YVQamSwV3g
```
Comment (ScriptAutomate) [inline README.rst:183]:
```suggestion
.. _Salt Project Community Discord: https://discord.gg/bPah23K7mG
```

File: SUPPORT.rst [hunk 7]
```diff
@@ -8,10 +8,10 @@ it may take a few moments for someone to reply.
 
 `<https://web.libera.chat/#salt>`_
 
-**SaltStack Slack** - Alongside IRC is our SaltStack Community Slack for the
+**SaltStack Slack** - Alongside IRC is our SaltStack Community Discord for the
 SaltStack Working groups. Use the following link to request an invitation.
 
-`<https://via.vmw.com/salt-slack>`_
+`<https://discord.gg/YVQamSwV3g>`_
```
Comment (ScriptAutomate) [inline SUPPORT.rst:14]:
```suggestion
`<https://discord.gg/bPah23K7mG>`_
```

File: doc/_incl/requisite_incl.rst [hunk 8]
```diff
@@ -7,4 +7,4 @@ following the instructions in the
     The Salt Project community can help offer advice and help troubleshoot
     technical issues as you're learning about Salt. One of the best places to
     talk to the community is on the
-    `Salt Project Slack workspace <https://saltstackcommunity.slack.com/>`_.
+    `Salt Project Discord Community <https://discord.gg/YVQamSwV3g>`_.
```
Comment (ScriptAutomate) [inline doc/_incl/requisite_incl.rst:10]:
```suggestion
    `Salt Project Discord Community <https://discord.gg/bPah23K7mG>`_.
```

File: doc/conf.py [hunk 9]
```diff
@@ -181,7 +181,7 @@
 .. _`salt-users`: https://groups.google.com/forum/#!forum/salt-users
 .. _`salt-announce`: https://groups.google.com/forum/#!forum/salt-announce
 .. _`salt-packagers`: https://groups.google.com/forum/#!forum/salt-packagers
-.. _`salt-slack`: https://via.vmw.com/salt-slack
+.. _`salt-discord`: https://discord.gg/YVQamSwV3g
```
Comment (ScriptAutomate) [inline doc/conf.py:184]:
```suggestion
.. _`salt-discord`: https://discord.gg/bPah23K7mG
```

File: doc/security/index.rst [hunk 10]
```diff
@@ -172,4 +172,4 @@ In addition to the mailing lists, SaltStack also provides the following resource
 
 * `SaltStack Security Announcements <https://www.saltstack.com/security-announcements/>`__ landing page
 * `SaltStack Security RSS Feed <http://www.saltstack.com/feed/?post_type=security>`__
-* `SaltStack Community Slack Workspace <http://saltstackcommunity.slack.com/>`__
+* `Salt Project Discord Community <https://discord.gg/YVQamSwV3g>`__
```
Comment (ScriptAutomate) [inline doc/security/index.rst:175]:
```suggestion
* `Salt Project Discord Community <https://discord.gg/bPah23K7mG>`__
```

File: doc/topics/tutorials/states_pt4.rst [hunk 11]
```diff
@@ -211,7 +211,7 @@ can be found on GitHub in the `saltstack-formulas`_ collection of repositories.
 If you have any questions, suggestions, or just want to chat with other people
 who are using Salt, we have a very active community and we'd love to hear from
 you. One of the best places to talk to the community is on the
-`Salt Project Slack workspace <https://saltstackcommunity.slack.com/>`_.
+`Salt Project Discord Community <https://discord.gg/YVQamSwV3g>`_.
```
Comment (ScriptAutomate) [inline doc/topics/tutorials/states_pt4.rst:214]:
```suggestion
`Salt Project Discord Community <https://discord.gg/bPah23K7mG>`_.
```

**Review by ScriptAutomate — COMMENTED** [review]
We are in the midst of replacing the Discord invite link to be from Chunga, even though this invite link will continue to work as a non-expiring link. Using this one to be part of the official documentation and the blog.

- Old non-expiring link (will continue to be available), but is by Derek: https://discord.gg/YVQamSwV3g
- New non-expiring link, but is by the community manager (Chunga): https://discord.gg/bPah23K7mG

The website is also being updated today. I'll apply a batch commit to make the changes.

**Review by ScriptAutomate — APPROVED** [review]

**Review by dmurphy18 — DISMISSED** [review]

**Review by ScriptAutomate — CHANGES_REQUESTED** [review]
> We are in the midst of replacing the Discord invite link to be from Chunga, even though this invite link will continue to work as a non-expiring link. Using this one to be part of the official documentation and the blog.
>
> - Old non-expiring link (will continue to be available), but is by Derek: https://discord.gg/YVQamSwV3g
> - New non-expiring link, but is by the community manager (Chunga): https://discord.gg/bPah23K7mG
>
> The website is also being updated today. I'll apply a batch commit to make the changes.

Going to wait on this at the moment. Need to see if we have a better solution this week for a permanent link that could be generated by something like a bot instead of specific user.

**Comment by dmurphy18** [comment]
@max-arnold Sorry forgot about this when I did PR https://github.com/saltstack/salt/pull/66850. Recommend doing this against the 3006.x branch, as it will be merged-forward to the other branches

**Comment by dmurphy18** [comment]
@max-arnold Suggest using lionk  https://discord.com/invite/J7b7EscrAs for the Discord invite, was used in last week's Open Hour and is not based on Chunga but Salt Project Admin

**Comment by max-arnold** [comment]
@dmurphy18 Rebased against 3006.x and used the proper Discord link

**Review by twangboy — APPROVED** [review]

**Review by dmurphy18 — APPROVED** [review]

**Comment by dmurphy18** [comment]
@ScriptAutomate Can you review this and help get it over the line, looks good to me

**Review by ScriptAutomate — APPROVED** [review]

**Pull request merged**